docs/CONTRIBUTING.md

```diff
@@ -7,29 +7,134 @@ sure you follow the steps below:
 
-## Developing environment
+For more information on how to get started, read on!
 
-After installing all dependencies with `yarn`, you can run
+## Setting Up Your Build Environment
 
-```sh
-yarn dev
-```
+To build **Vivify** you need the following dependencies:-
 
-to run the development server for Vivify on port 3000 that will (1)
-automatically reload when you make changes to the code, and (2), unlike the
-installed version, not shut down when there are no connected clients.
+**MacOS:**
 
-With the development server running, use
+    TODO: Mac instructions
 
-```sh
-yarn viv
-```
+**Arch Linux/Manjaro:**
 
-instead of your installed Vivify executable. This will (1) connect to the
-development server on port 3000 instead of running the installed server, and (2)
-use the `viv` executable in the repository.
+    sudo pacman -S --needed yarn make zip
+
+**Fedora and derivatives:**
+
+    sudo dnf install yarn make zip
+
+> [!NOTE]
+> The version of **Node.js** shipped with some Linux distributions will fail to
+> build **Vivify**, see the section below to install the latest node from **nvm**
+
+## Installing Node.js
+
+Node.js may be available with your OS or in your distros package manager.
+Alternatively you can install the latest version of Node.js using **[Node
+Version Manager](https://github.com/nvm-sh/nvm)** (**nvm**)
+
+To install **nvm** use:-
+
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
+
+You will then need to reload your terminal by either closing it and opening a
+new one or by re-sourcing your run commands (bash eg: `source ~/.bashrc`)
+
+Now you can install the latest version of Node.js:-
+
+    nvm install node
+
+> [!TIP]
+> You can return to the system version of node using `nvm use system`
+>
+> For more details on nvm usage or for troubleshooting installing nvm, visit the
+> nvm [Documentation][1] page
+>
+> [1]:<https://github.com/nvm-sh/nvm?tab=readme-ov-file#node-version-manager--->
+
+## Building Vivify
+
+First clone and open the **Vivify** repository
+
+    git clone https://github.com/jannis-baum/Vivify.git
+    cd Vivify
+
+Run `yarn` to download all node dependencies
+
+    yarn
+
+Then build **Vivify** by running `make <OS-family>` where *OS-family* is your
+operating system (Linux or MacOS)
+
+**MacOS:**
+
+    make macos
+
+**Linux:**
+
+    make linux
+
+If you want to install **Vivify** for use outside of your development
+environmentm you can use `make install` but first you need to run the
+`configure <install-dir>` script, eg:-
+
+    ./configure ~/.local/bin
+    make install
+
+Once installed you can launch **viv** by calling `viv <some-path-or-file>`
+
+> [!TIP]  
+> Ideally the install location should be included in your $PATH variable
+
+## Running Vivify for Development
+
+Once you have built **Vivify** or installed it using the pre-built binaries, you
+can then run the **Vivify** server in development mode to try it out with
+live-updates as you are coding.
```

Comment:
This is not quite accurate: You don't need to build Vivify to run it in development mode (just having `yarn` installed, cloning the repo, and running `yarn` one time is enough), and if you installed it using pre-built binaries you can't run development mode at all.